From eebfc0c799906fb69383660b4871105cc2ab6524 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh+mozilla@glandium.org>
Date: Tue, 9 Aug 2022 22:42:44 +0200
Subject: [PATCH 2/2] Bug 1782988 - Fix use of arc4random_buf use in ping.cpp.

The code was probably never built before glibc 2.36, because before
that, only Android and some BSDs had arc4random_buf, but none of those
actually built this code.
---
 toolkit/crashreporter/client/ping.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/crashreporter/client/ping.cpp b/toolkit/crashreporter/client/ping.cpp
index e3839e6475db..27fe2e829516 100644
--- a/toolkit/crashreporter/client/ping.cpp
+++ b/toolkit/crashreporter/client/ping.cpp
@@ -53,7 +53,7 @@ GenerateUUID()
 
   CFRelease(uuid);
 #elif defined(HAVE_ARC4RANDOM_BUF) // Android, BSD, ...
-  arc4random_buf(id, sizeof(UUID));
+  arc4random_buf(&id, sizeof(UUID));
 #else // Linux
   int fd = open("/dev/urandom", O_RDONLY);
 
-- 
2.37.1

